packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/assetModelDataStreamExplorer/assetModelDataStreamExplorer.tsx

```diff
@@ -0,0 +1,246 @@
+import { useModelBasedWidgets } from './useModelBasedWidgets';
+import { styledQueryWidgetOnDrop } from '../../useQuery';
+import { QueryProperties } from '~/customization/widgets/types';
+import { assignDefaultStyles } from '~/customization/widgets/utils/assignDefaultStyleSettings';
+import { IoTSiteWiseDataStreamQuery } from '~/types';
+
+export interface AssetModelDataStreamExplorerProps {
+  client: IoTSiteWiseClient;
+}
+
+export const AssetModelDataStreamExplorer = ({ client }: AssetModelDataStreamExplorerProps) => {
+  const { widgets, update } = useModelBasedWidgets();
+  const dashboardConfiguration = useSelector((dashboardState: DashboardState) => dashboardState.dashboardConfiguration);
+  const selection = useSelection({ filter: isQueryWidget });
+
+  // const widgetTypesInSelection = selection?.types ?? [];
+  // xy-plot widgets are styled query objects
+  const t = selection?.type;
+
+  const [propertiesMaybe, setProperties] =
+    selection?.useProperty(
+      (properties) => properties,
+      (_properties, updatedProperties) => updatedProperties
+    ) ?? [];
+  const defaultQuery: QueryProperties = {
+    queryConfig: {
+      source: 'iotsitewise',
+      query: undefined,
+    },
+  };
+  const properties = propertiesMaybe ? maybeWithDefault(defaultQuery, propertiesMaybe) ?? defaultQuery : defaultQuery;
+
+  const setAssetModels = (updatedAssetModels: AssetModelQuery[] | undefined) => {
+    if (!setProperties || !t || !isJust(t)) return;
+
+    let updatedProperties = { ...properties };
+    const compositeWidgetForAggregationInformation = {
+      type: t.value,
+      properties,
+    } as QueryConfigWidget;
+    // hangle styled widget
+    if (t.value === 'xy-plot') {
+      const styledQuery = styledQueryWidgetOnDrop(
+        { assetModels: updatedAssetModels },
+        compositeWidgetForAggregationInformation
+      );
+      updatedProperties = {
+        ...properties,
+        queryConfig: {
+          ...properties.queryConfig,
+          query: {
+            ...properties.queryConfig.query,
+            assetModels: (styledQuery as unknown as IoTSiteWiseDataStreamQuery).assetModels,
+          },
+        },
+      };
+    } else {
+      const styledProperties = assignDefaultStyles({
+        ...compositeWidgetForAggregationInformation,
+        properties: {
+          ...compositeWidgetForAggregationInformation.properties,
+          queryConfig: {
+            ...compositeWidgetForAggregationInformation.properties.queryConfig,
+            query: {
+              ...compositeWidgetForAggregationInformation.properties.queryConfig.query,
+              assetModels: updatedAssetModels,
+            },
+          },
+        },
+      }).properties;
+      updatedProperties = {
+        ...styledProperties,
+      };
+    }
+
+    setProperties(updatedProperties);
+  };
+
+  const definedAssetModels = properties.queryConfig.query?.assetModels ?? [];
+
+  const onReset = useCallback(() => {
```

Comment:
Why do we need to optimize the functions with useCallback?

Reply:
removed, the optimization is required on some of the other functions defined adjacent to this. This one was added during debugging the performance issue.